### ISSUE TITLE
ansible-test integration in Docker fails with error about ansible_default_ipv4

### ISSUE THREAD

**Issue by iamjpotts** [ansible-collections/community.docker#503]
##### SUMMARY

1. Checkout main
2. Create a virtualenv with Python 3.8, ansible, docker, and pytest packages
3. Activate virtualenv
4. Run `ansible-test integration --docker --python 3.8`

Note: The virtualenv is only to install get the `ansible-test` command on the docker host; believe the rest is containerized.

Fails, notice `ansible_default_ipv4`:

```
TASK [docker_config : Create a Swarm cluster] **********************************
fatal: [testhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'ansible_default_ipv4' is undefined\n\nThe error appears to be in '/root/ansible_collections/community/docker/tests/output/.tmp/integration/docker_config-7sm83aml-ÅÑŚÌβŁÈ/tests/integration/targets/docker_config/tasks/test_docker_config.yml': line 16, column 5, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n  - name: Create a Swarm cluster\n    ^ here\n"}
```

`$ pip list`

```
Package            Version
------------------ ---------
ansible            6.6.0
ansible-core       2.13.6
attrs              22.1.0
certifi            2022.9.24
cffi               1.15.1
charset-normalizer 2.1.1
cryptography       38.0.3
docker             6.0.1
idna               3.4
iniconfig          1.1.1
Jinja2             3.1.2
MarkupSafe         2.1.1
packaging          21.3
pip                22.3.1
pluggy             1.0.0
py                 1.11.0
pycparser          2.21
pyparsing          3.0.9
pytest             6.2.5
PyYAML             6.0
requests           2.28.1
resolvelib         0.8.1
setuptools         60.2.0
toml               0.10.2
typing             3.7.4.3
urllib3            1.26.13
websocket-client   1.4.2
wheel              0.37.1
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Containerized integration tests run with `ansible-test`

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.6]
  config file = None
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = ~/dev/ansible/collections/venv/lib/python3.8/site-packages/ansible
  ansible collection location = ~/.ansible/collections:/usr/share/ansible/collections
  executable location = ~/dev/ansible/collections/venv/bin/ansible
  python version = 3.8.10 (default, Jun 22 2022, 20:18:18) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True

```

##### COLLECTION VERSION

```paste below
ansible-galaxy collection list community.docker

# ~/dev/ansible/collections/venv/lib/python3.8/site-packages/ansible_collections
Collection       Version
---------------- -------
community.docker 2.7.1 
```

##### CONFIGURATION

Output is empty

##### OS / ENVIRONMENT

`Ubuntu 20.04`
`Docker version 20.10.21, build baeda1f`

##### STEPS TO REPRODUCE

Run `ansible-test integration --docker --python 3.8`

##### EXPECTED RESULTS

Containerized tests should pass.

##### ACTUAL RESULTS

Fails with output in summary above.

**Comment by felixfontein** [comment]
The default container is not supported for integration tests in this collection.

**Comment by iamjpotts** [comment]
Is there a containerized alternative?

**Comment by felixfontein** [comment]
Yes, basically every OS container except the default container should work. `ansible-test integration --help` should list its supported containers. For me it prints:
```
target docker images and supported python version (choose one):
  base (3.11, 2.7, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10)
  default (3.11, 2.7, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10)
  alpine3 (3.10)
  centos7 (2.7)
  fedora36 (3.10)
  opensuse15 (3.6)
  ubuntu2004 (3.8)
  ubuntu2204 (3.10)
  {image}  # python must be specified for custom images
```
So I can use `--docker alpine3`, `--docker centos7`, `--docker opensuse15`, `--docker ubuntu2004`, and `--docker ubuntu2204`.

**Comment by iamjpotts** [comment]
Tried some other combinations, still getting errors.

Example 1, using Ubuntu 20.04:

```
$ ansible-test integration --docker ubuntu2004
```

...results in this (notice similar failure point near things related to `swarm` compared to `default`):

```
TASK [docker_config : shell] ***************************************************
changed: [testhost]

TASK [docker_config : Create a Swarm cluster] **********************************
fatal: [testhost]: FAILED! => {"changed": false, "msg": "Can not create a new Swarm Cluster: 400 Client Error for http+docker://localhost/v1.41/swarm/init: Bad Request (\"must specify a listening address because the address to advertise is not recognized as a system address, and a system's IP address to use could not be uniquely identified\")"}

TASK [docker_config : Remove a Swarm cluster] **********************************
ok: [testhost]

RUNNING HANDLER [setup_remote_tmp_dir : delete temporary directory] ************
included: /root/ansible_collections/community/docker/tests/output/.tmp/integration/docker_config-krtt2tev-ÅÑŚÌβŁÈ/tests/integration/targets/setup_remote_tmp_dir/tasks/default-cleanup.yml for testhost

RUNNING HANDLER [setup_remote_tmp_dir : delete temporary directory] ************
changed: [testhost]

PLAY RECAP *********************************************************************
testhost                   : ok=43   changed=21   unreachable=0    failed=1    skipped=6    rescued=0    ignored=0   

NOTICE: To resume at this test target, use the option: --start-at docker_config
NOTICE: To resume after this test target, use the option: --start-at docker_container
FATAL: Command "ansible-playbook docker_config-ilu4our7.yml -i inventory" returned exit status 2.
FATAL: Command "docker exec ansible-test-controller-WQKMAm8r /usr/bin/env ANSIBLE_TEST_CONTENT_ROOT=/root/ansible_collections/community/docker LC_ALL=en_US.UTF-8 /usr/bin/python3.8 /root/ansible/bin/ansible-test integration --allow-destructive --containers '{}' --truncate 362 --color yes --host-path tests/output/.tmp/host-viqwuzc4 --metadata tests/output/.tmp/metadata-ts9urodx.json --exclude connection_docker --exclude connection_docker_api --exclude connection_nsenter --exclude generic_ssh_connection --exclude inventory_docker_machine" returned exit status 1.
```

Example 2, attempting to exclude the failing target:

```
$ ansible-test integration --docker ubuntu2004 --include docker_image --exclude docker_config
```

...results in:

```
ASK [setup_docker_registry : fail] ********************************************
skipping: [testhost]

TASK [setup_docker_registry : Start test registry] *****************************
changed: [testhost]

TASK [setup_docker_registry : Get registry URL] ********************************
ok: [testhost]

TASK [setup_docker_registry : include_tasks] ***********************************
ERROR! couldn't resolve module/action 'community.crypto.openssl_privatekey'. This often indicates a misspelling, missing collection, or incorrect module path.

The error appears to be in '/root/ansible_collections/community/docker/tests/output/.tmp/integration/docker_image-2o8h4827-ÅÑŚÌβŁÈ/tests/integration/targets/setup_docker_registry/tasks/setup-frontend.yml': line 52, column 5, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


  - name: Create private key for frontend certificate
    ^ here
NOTICE: To resume at this test target, use the option: --start-at docker_image
NOTICE: To resume after this test target, use the option: --start-at docker_image_info
FATAL: Command "ansible-playbook docker_image-k8ueuvdr.yml -i inventory" returned exit status 4.
FATAL: Command "docker exec ansible-test-controller-uSUo9Jd7 /usr/bin/env ANSIBLE_TEST_CONTENT_ROOT=/root/ansible_collections/community/docker LC_ALL=en_US.UTF-8 /usr/bin/python3.8 /root/ansible/bin/ansible-test integration --allow-destructive --containers '{}' --include docker_image --truncate 362 --color yes --host-path tests/output/.tmp/host-trluib78 --metadata tests/output/.tmp/metadata-kj4salgj.json --exclude connection_docker --exclude connection_docker_api --exclude connection_nsenter --exclude docker_config --exclude generic_ssh_connection --exclude inventory_docker_machine" returned exit status 1.
```

**Comment by felixfontein** [comment]
> ERROR! couldn't resolve module/action 'community.crypto.openssl_privatekey'. This often indicates a misspelling, missing collection, or incorrect module path.

You need to install the required collections in the same path structure to be able to run the tests. See tests/requirements.yml for which collections are needed.

**Comment by felixfontein** [comment]
(Also why do you want to run the docker_config tests? Your PR only affects docker_image.)

**Comment by iamjpotts** [comment]
> (Also why do you want to run the docker_config tests? Your PR only affects docker_image.)

I prefer to run all the tests first to make sure they're all green before making any changes.

Not being broadly familiar with the code base, running everything seemed like the right approach over a targeted approach.

**Comment by felixfontein** [comment]
Should be fixed by #514.

**Comment by felixfontein** [comment]
> Not being broadly familiar with the code base, running everything seemed like the right approach over a targeted approach.

Running everything is somewhat dangerous since the tests use the Docker daemon on the host and reconfigure it multiple times (for the Docker Swarm tests), this can easily make Docker daemon unstable (which is why the tests in CI are split up into five different groups, where three groups are reserved for Swarm tests and only two for everything else).
